File: batches/batch_11_partial_copy/compare.py
"""
Comparison - Partial Copy (CHEATING)
Only core logic copied: filter and write. Different I/O, different structure.
"""


def process_data():
    data = fetch_from_db()
    filtered = [x for x in data if len(str(x).strip()) > 5]
    save_results(filtered)
    return len(filtered)


def fetch_from_db():
    with open("input.txt", "r") as f:
        return f.readlines()


def save_results(items):
    with open("output.txt", "w") as f:
        f.write("\n".join(str(i).strip() for i in items))


if __name__ == "__main__":
    process_data()

File: batches_javascript/batch_11_partial_copy/compare.js
const fs = require('fs');

function main() {
    const lines = fs.readFileSync('input.txt', 'utf8').split('\n').filter(l => l.trim().length > 5).map(l => l.trim());
    fs.writeFileSync('output.txt', lines.join('\n'));
    console.log(lines.length);
}

main();
